### PR TITLE
Project

### PR DESCRIPTION
**Pull request by ger619**
### **Project requirements**
API First you need to choose an API to base the development of the webapp on. The API should allow you to get numeric values based on a parameter. We recommend that you choose an API that is open or needs an API key. If you choose an API that require authentication, you should implement it on your own.

### Some example APIs are:
- [Financial modeling prep API](https://financialmodelingprep.com/developer/docs/): stock data.

- [Air pollution API](https://openweathermap.org/api/air-pollution): air quality data.

You can find many APIs in [this GitHub repo](https://github.com/public-apis/public-apis) or in [ProgrammableWeb](https://www.programmableweb.com/category/all/apis). Some of the APIs requires a token for authentication and some others are just open.

### Design

- You should follow these [design guidelines](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)), including:

    - Colors (select one main color for your website).
    - Typography: font face, size and weight.
    - Layout: composition and space between elements.
    
Original design idea by [Nelson Sakwa on Behance](https://www.behance.net/sakwadesignstudio).

The [Creative Commons license of the design](https://creativecommons.org/licenses/by-nc/4.0/) requires that you give appropriate credit to the author. Therefore, you must do it in the README of your project.

**Interactions**

- Home page

  - When the page loads, the webapp shows the list of categories that could be filtered out by some parameter, for example by the category name.
  - Along with the category name, you will display some numeric values per category that come from the API.
  - When the user clicks (or taps) on a category item, the application navigates to the details page.

**Details page**

- In the details page, the webapp retrieves data from the API to show detailed data in the category.
- When the user clicks on the "Back" button (<), the user navigates to the home page.

**Testing requirements**

- Create unit tests for pure functions in your Redux code.
- Create integration tests for your application using the React Testing Library.
    - You may need to mock the access to the API, so that your tests don't send actual requests.
    - You may need to mock the connection to the Redux Store.

**Technical requirements**

- The project is a single-page application (SPA) built with React and Redux.
- The data retrieved from the API should be stored in the Redux store.
- You should filter the data that you retrieve from the API using a Filter stateless component.
- Every page (the main page and the pages for each item) should have a unique route within the SPA.
- The project should be deployed and accessible online (add a link to the online version of your app to your README file).

### CODE REVIEW

File: src/components/Home.jsx [hunk 1]
```diff
@@ -3,23 +3,37 @@ import {
...
       <Title level={2} className="heading"> Ger Crypto Stats</Title>
       <Row>
         <Col span={12}><Statistic title="All Crypto Currencies" value={globalStats.total} /></Col>
-        <Col span={12}><Statistic title="All Crypto Currencies" value={millify(globalStats.totalExchanges)} /></Col>
-        <Col span={12}><Statistic title="All Crypto Currencies" value="5" /></Col>
-        <Col span={12}><Statistic title="All Crypto Currencies" value="5" /></Col>
+        <Col span={12}><Statistic title="Total Exchanges" value={globalStats.totalExchanges} /></Col>
+        <Col span={12}><Statistic title="Total Market Cap" value={millify(globalStats.totalMarketCap)} /></Col>
+        <Col span={12}><Statistic title="Total Volume" value={millify(globalStats.total24hVolume)} /></Col>
+        <Col span={12}><Statistic title="Total Markets" value={millify(globalStats.totalMarkets)} /></Col>
```
Comment (Kingstalux) [inline src/components/Home.jsx]: - [ ] Good job so far 👍 Kindly apply the same styling and colors as found in the original design. Pay special attention to the fact that in the mobile view, the items should be 2 in a row and the colors should be alternating 💯 
- Your design:
Annotation 2022-06-24 164301
- Required design:
Annotation 2022-06-24 164326

Kindly do this for both the home and the details pages 👍

File: src/test/components.test.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      .toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+  test('test AllCrypto', () => {
+    const tree = renderer
+      .create(<Provider store={store}><AllCrypto /></Provider>)
+      .toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+  test('test CryptoDetails', () => {
```
Comment (Kingstalux) [inline src/test/components.test.js:22]: - [ ] Good job writing tests for your components 💯 Kindly fix your failing tests
- Your site:
Annotation 2022-06-24 165114

**Review by Kingstalux — CHANGES_REQUESTED** [review]
Hi @ger619, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
68747470733a2f2f6d656469612e67697068792e636f6d2f6d656469612f6e59597451644c7177776c54653363634b682f67697068792e676966


## Highlights 👏 
✅ App has the correct functionality 🔥 
✅ Good commit history 👍 
✅ Linters are passing 👌 
✅ Great PR description 👏 



## Required Changes ♻️
_Check the comments under the review._
- [ ] Kindly record a video of maximum 5 minutes in which you will present the key features of your site and some piece of interesting code you worked on in this project 👍
- [ ] Your README should contain an "Acknowledgements" section or similar that gives appropriate credit to [Nelson Sakwa](https://www.behance.net/sakwadesignstudio) the [author of the original design](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)), as required by the [Creative Commons License](https://creativecommons.org/licenses/) 💯 
 
## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/test/components.test.js (hunk 2 above)
Comment (Davidosky007) [reply src/test/components.test.js:22]: * Addresssed 👌

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @ger619 👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
* Kindly record a video of maximum 5 minutes in which you will present the key features of your site and some piece of interesting code you worked on in this project 👍
* Your README should contain an "Acknowledgements" section or similar that gives appropriate credit to [Nelson Sakwa](https://www.behance.net/sakwadesignstudio) the [author of the original design](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)), as required by the [Creative Commons License](https://creativecommons.org/licenses/) 💯
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by KDlamini — APPROVED** [review]
### Hi @ger619 🙋‍♀️,  

Excellent work! 💯 
You have done a great job implementing all the required changes. 
Your feature is complete! There is nothing else to say other than... it's time to merge it. :shipit: 
Congratulations!!!🎊 🥳 

## Highlights 🌞 
✔️  Good PR description
✔️  Redux state management used correctly
✔️  Good use of React components, Routing, and Hooks
✔️  Commit messages are meaningful
✔️  No linter errors

# Status: Approved ✔️ 

giphyapproved
 
Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @KDlamini  in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**